### ISSUE TITLE
[BUG] error location for function type in sub-expression in element content 

### ISSUE THREAD

**Issue by line-o** [eXist-db/exist#3623]
**Describe the bug**

The location of an expression evaluating to a function type inside of a sub-expression inside of element content must be reported.

Example:

```xquery
element test { true(), map{} }
```

**Expected behavior**

Throws an error at runtime but without location information.

**To Reproduce**

There are already pending tests in place in #3550, which adds location to the first set of errors of this type.

**Comment by joewiz** [comment]
The example above is missing the enclosed content expression required by the computed element constructor. To match the description of the issue, the test would need an enclosed content expression. For example:

```xquery
element foo { map{} }
```

The error raised by this revised query is:

> err:XQTY0105 Enclosed expression contains function item

The references pending tests can be found here:

https://github.com/eXist-db/exist/blob/8d2dcbccde87de5adcedfd34b4aad033fa78a6e8/exist-core/src/test/xquery/errors.xql#L328-L360.

**Comment by line-o** [comment]
> The example above is missing the enclosed content expression required by the computed element constructor. 
@joewiz of course! That was just a type, I changed it in the description. Thanks for pointing it out.

**Comment by line-o** [comment]
fixed by c553667221306f4df149c622e0e4f2074052da60
